app-editors/atom/atom-1.7.3.ebuild

```diff
@@ -5,81 +5,363 @@
 
-src_prepare(){
+get_install_suffix() {
+	local c=(${SLOT//\// })
+	local slot=${c[0]}
+	local suffix
+
+	if [[ "${slot}" == "0" ]]; then
+		suffix=""
+	else
+		suffix="-${slot}"
+	fi
+
+	echo -n "${suffix}"
+}
+
+get_install_dir() {
+	echo -n "/usr/$(get_libdir)/atom$(get_install_suffix)"
+}
+
+package_dir() {
+	local binmod="${1}" binmod_v
+	eval binmod_v=\${$(echo ${binmod//-/_}_V | tr '[:lower:]' '[:upper:]')}
+	echo -n ${binmod}-${binmod_v}
+}
+
+_unpack_npm_package() {
+	local a="${1}" basename suffix
+
+	basename=${a%.*}
+	suffix=${basename##*.}
+	if [[ "${suffix}" == "tar" ]]; then
+		basename=${basename%.*}
+	fi
+
+	unpack ${a}
+	mv package "${basename}" || die
+}
+
+_have_patches_for() {
+	local _patches="${1}-*.patch" _find
+	_find=$(find "${FILESDIR}" -maxdepth 1 -name "${_patches}" -print -quit)
+	test -n "$_find"
+}
+
+src_unpack() {
+	local a
+
+	for a in ${A} ; do
+		case ${a} in
+		*.rpm) srcrpm_unpack "${a}" ;;
+		nodegit-promise*|node-lodash*) _unpack_npm_package "${a}" ;;
+		*) unpack "${a}" ;;
+		esac
+	done
+
+	mkdir "${WORKDIR}/atom-${MY_PV}" || die
+	mv "${WORKDIR}/usr" "${WORKDIR}/atom-${MY_PV}" || die
+}
+
+src_prepare() {
+	local install_dir="$(get_install_dir)"
+	local patch binmod _s nan_s="${WORKDIR}/nan-${NAN_V}"
+
+	cd "${S}/usr/share/atom/resources/app" || die
 	epatch "${FILESDIR}/${PN}-python.patch"
-	sed -i  -e "/exception-reporting/d" \
-		-e "/metrics/d" package.json
-	sed -e "s/<%= description %>/$pkgdesc/" \
-		-e "s|<%= installDir %>/share/<%= appFileName %>/atom|/usr/bin/atom|"\
-		-e "s|<%= iconPath %>|atom|"\
-		-e "s|<%= appName %>|Atom|" \
-		resources/linux/atom.desktop.in > resources/linux/Atom.desktop
-
-	# Fix atom location guessing
-	sed -i -e 's/ATOM_PATH="$USR_DIRECTORY\/share\/atom/ATOM_PATH="$USR_DIRECTORY\/../g' \
+	epatch "${FILESDIR}/${PN}-unbundle-electron.patch"
+
+	sed -i -e "s|{{ATOM_PATH}}|/usr/$(get_libdir)/electron/electron|g" \
 		./atom.sh \
-		|| die "Fail fixing atom-shell directory"
+		|| die
 
-	# Make bootstrap process more verbose
-	sed -i -e 's@node script/bootstrap@node script/bootstrap --no-quiet@g' \
-		./script/build \
-		|| die "Fail fixing verbosity of script/build"
-	default
+	sed -i -e "s|{{ATOM_RESOURCE_PATH}}|${install_dir}/app.asar|g" \
+		./atom.sh \
+		|| die
+
+	local env="export NPM_CONFIG_NODEDIR=/usr/include/electron/node/"
+	sed -i -e \
+		"s|\"\$binDir/\$nodeBin\" --harmony_collections|${env}\nexec /usr/bin/node|g" \
+			apm/bin/apm || die
+
+	rm apm/bin/node || die
+
+	sed -i -e "s|/usr/share/atom/atom|/usr/bin/atom|g" \
+		"${S}/usr/share/applications/atom.desktop" || die
+
+	cd "${S}" || die
+
+	for binmod in ${BINMODS}; do
+		_s="${WORKDIR}/$(package_dir ${binmod})"
+		cd "${_s}" || die
+		if _have_patches_for "${binmod}"; then
+			for patch in "${FILESDIR}"/${binmod}-*.patch; do
+				epatch "${patch}"
+			done
+		fi
+	done
+
+	# Unbundle bundled libs from modules
+
+	_s="${WORKDIR}/$(package_dir git-utils)"
+	python "${FILESDIR}/gyp-unbundle.py" \
```

Comment:
No no no. That's not how you should invoke a Python interpreter. See the ${EPYTHON} variable in https://github.com/gentoo/gentoo/blob/master/eclass/python-utils-r1.eclass#L155.